src/test/java/hudson/plugins/git/CredentialsUserRemoteConfigTest.java

```diff
@@ -0,0 +1,145 @@
+                break;
+            }
+        }
+        assertThat("The system credentials provider is enabled", store, notNullValue());
+    }
+
+    @Issue("JENKINS-30515")
+    @Test
+    public void checkoutWithValidCredentials() throws Exception {
+        sampleRepo.init();
+        store.addCredentials(Domain.global(), createCredential(CredentialsScope.GLOBAL, "github"));
+        store.save();
+
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n"
+                        + "  checkout(\n"
+                        + "    [$class: 'GitSCM', \n"
+                        + "      userRemoteConfigs: [[credentialsId: 'github', url: $/" + sampleRepo + "/$, branches: [[name: '*/master']], doGenerateSubmoduleConfigurations: false]]]\n"
+                        + "  )"
+                        + "}"));
+        WorkflowRun b = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        r.assertLogContains("using credential github", b);
+    }
+
+    @Issue("JENKINS-30515")
+    @Test
+    public void checkoutWithDifferentCredentials() throws Exception {
+        sampleRepo.init();
+        store.addCredentials(Domain.global(), createCredential(CredentialsScope.GLOBAL, "other"));
+        store.save();
+
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n"
+                        + "  checkout(\n"
+                        + "    [$class: 'GitSCM', \n"
+                        + "      userRemoteConfigs: [[credentialsId: 'github', url: $/" + sampleRepo + "/$, branches: [[name: '*/master']], doGenerateSubmoduleConfigurations: false]]]\n"
+                        + "  )"
+                        + "}"));
+        WorkflowRun b = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        System.out.println(JenkinsRule.getLog(b));
+        r.assertLogContains("Warning: CredentialId \"github\" could not be found", b);
+    }
+
+    @Issue("JENKINS-30515")
+    @Test
+    public void checkoutWithVInvalidCredentials() throws Exception {
+        sampleRepo.init();
+        store.addCredentials(Domain.global(), createCredential(CredentialsScope.SYSTEM, "github"));
+        store.save();
+
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n"
+                        + "  checkout(\n"
+                        + "    [$class: 'GitSCM', \n"
+                        + "      userRemoteConfigs: [[credentialsId: 'github', url: $/" + sampleRepo + "/$, branches: [[name: '*/master']], doGenerateSubmoduleConfigurations: false]]]\n"
+                        + "  )"
+                        + "}"));
+        WorkflowRun b = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        r.assertLogContains("Warning: CredentialId \"github\" could not be found", b);
```

Comment:
This seems (to me) to be asserting the wrong message.

The test creates a credential named "github" but then asserts that the message appears that the credential "github" is not found.  Shouldn't the credential be found in this case?